app/controllers/admin/routes.py

```diff
@@ -153,6 +156,54 @@ def rsvpLogDisplay(eventId):
     else:
         abort(403)
 
+@admin_bp.route('/event/<eventId>/renew', methods=['POST'])
+def renewEvent(eventId):
+    try: 
+        formData = request.form
+        try:
+            assert formData['timeStart'] < formData['timeEnd']
+        except AssertionError:
+            flash("End time must be after start time", 'warning')
+            return redirect(url_for('admin.eventDisplay', eventId = eventId))
+        
+        try:
+            if formData.get('dateEnd'):
+                assert formData['dateStart'] < formData['dateEnd']
+        except AssertionError:
+            flash("End date must be after start date", 'warning')
+            return redirect(url_for('admin.eventDisplay', eventId = eventId))
+
+
+        priorEvent = model_to_dict(Event.get_by_id(eventId))
+        newEventDict = priorEvent.copy()
+        newEventDict.pop('id')
+        newEventDict.update({
+                    'program': int(priorEvent['program']['id']),
+                    'term': int(priorEvent['term']['id']),
+                    'timeStart': formData['timeStart'],
+                    'timeEnd': formData['timeEnd'],
+                    'location': formData['location'],
+                    'startDate': f'{formData["startDate"][-4:]}-{formData["startDate"][0:-5]}',
+                    'endDate': f'{formData["endDate"][-4:]}-{formData["endDate"][0:-5]}',
+                    'isRecurring': bool(priorEvent['recurringId'])
+                    })
+        newEvent, message = attemptSaveEvent(newEventDict, renewedEvent = True)
+        if message:
+            flash(message, "danger")
+            return redirect(url_for('admin.eventDisplay', eventId = eventId))
+
+        copyRsvpToNewEvent(priorEvent, newEvent[0])
```

Comment:
You use this function but it is not imported and causes an error.

Reply:
Yes I messed up when resolving a merge conflict.